modules/PromiseUtils.js

```diff
@@ -0,0 +1,3 @@
+export function isPromise(obj) {
+  return obj && typeof obj.then === 'function'
+}
```

Comment:
Dunno if this is worth throwing in a module. Maybe move it to AsyncUtils at least?